### PR TITLE
bandwidth: 1.9.4 -> 1.10.1, enable on arm

### PR DESCRIPTION
**Pull request by r-burns**
add make flags needed for cross-compilation
tested on pkgsCross.{aarch64-multiplatform,armv7l-hf-multiplatform}

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 137154` at a7081e0f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>bandwidth</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/bandwidth/default.nix:45:3:
```
   |
45 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/tools/misc/bandwidth/default.nix [hunk 1]
```diff
@@ -6,21 +6,41 @@ let
...
+  + ''
+    sed -i 's,-m32,,g' OOC/Makefile
+  ''
+  # Fix wrong comment character
+  + ''
+    sed -i 's,# 32,; 32,g' routines-x86-32bit.asm
+  ''
+  # Fix missing symbol exports for macOS clang
+  + ''
+    echo global _VectorToVector128 >> routines-x86-64bit.asm
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/bandwidth/default.nix:29]:
```suggestion
    sed -i 's,ar ,$(AR) ,g' OOC/Makefile

    # Remove unnecessary -m32 for 32-bit targets
    sed -i 's,-m32,,g' OOC/Makefile

    # Fix wrong comment character
    sed -i 's,# 32,; 32,g' routines-x86-32bit.asm

    # Fix missing symbol exports for macOS clang
    echo global _VectorToVector128 >> routines-x86-64bit.asm
```

File: pkgs/tools/misc/bandwidth/default.nix [hunk 2]
```diff
@@ -31,7 +51,7 @@ stdenv.mkDerivation rec {
     homepage = "https://zsmith.co/bandwidth.html";
     description = "Artificial benchmark for identifying weaknesses in the memory subsystem";
     license = licenses.gpl2Plus;
-    platforms = platforms.x86;
+    platforms = platforms.x86 ++ platforms.arm ++ platforms.aarch64;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/bandwidth/default.nix:54]: This looks more like unix or all.

File: pkgs/tools/misc/bandwidth/default.nix (hunk 1 above)
Comment (r-burns) [reply pkgs/tools/misc/bandwidth/default.nix]: Done

File: pkgs/tools/misc/bandwidth/default.nix (hunk 2 above)
Comment (r-burns) [reply pkgs/tools/misc/bandwidth/default.nix:48]: No, each platform has its own assembly routines, so e.g. risc-v and power aren't supported

Comment (SuperSandro2000) [reply pkgs/tools/misc/bandwidth/default.nix:48]: risk-v and power have such a low support level that it probably can't even compile all the dependencies of bandwhich for those platforms. Generally we don't need to consider them at this point.

**Pull request merged**